python/tank/deploy/tank_commands/install.py

```diff
@@ -163,7 +196,8 @@ def _run(self, log, suppress_prompts, env_name, engine_instance_name, app_name):
         log.info("Installing into environment %s and engine %s." % (env_name, engine_instance_name))
     
         try:
-            env = self.tk.pipeline_configuration.get_environment(env_name)
+            env = self.tk.pipeline_configuration.get_environment(env_name, writable=True)
+            env.use_preserving_yaml_parser(preserve_yaml)
```

Comment:
yep i agree, `use_xxx(True)` or `enable_xxx(True)` are edge cases for me. But I agree. 

`set_preserving_yaml_parser()` however sounds to be like you are supposed to pass a yaml parser object as a parameter that the class should be using! 

I'll change it to `env.set_yaml_preserve_mode(preserve_yaml)`